### PR TITLE
Readd diesel store and implement the relevant changes

### PR DESCRIPTION
**Pull request by weiznich**
This readds the diesel store and implements the relevant changes for the expiry PR. Hopefully I've included all removed files. (Next time it might be easier to just leave them in place by just disabling the feature temporary)

### CODE REVIEW

File: examples/diesel-store-with-custom-table.rs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            ))
+            .execute(conn)?;
+        Ok(())
+    }
+
+    fn migrate(conn: &mut SqliteConnection) -> Result<(), DieselStoreError> {
+        // or create the table via normal diesel migrations on startup and leave that
+        // function empty
+        conn.batch_execute(
+            "CREATE TABLE `my_sessions` (`id` TEXT PRIMARY KEY NOT NULL, `expiration_time` TEXT \
```
Comment (maxcountryman) [inline examples/diesel-store-with-custom-table.rs:61]: I think this should be expiry_date instead of expiration_time.

File: src/diesel_store.rs [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+        qb.push_sql("( ");
+        qb.push_identifier(sessions::id::NAME)?;
+        // we need these hacks to not depend on all diesel backends on the same time
+        if connection_type.ends_with("Mysql") {
+            qb.push_sql(" CHAR(36) PRIMARY KEY NOT NULL, ");
+        } else {
+            qb.push_sql(" TEXT PRIMARY KEY NOT NULL, ");
+        }
+        qb.push_identifier(sessions::expiry_date::NAME)?;
+        qb.push_sql(" TIMESTAMP NULL, ");
```
Comment (maxcountryman) [inline src/diesel_store.rs:177]: This is no longer nullable.

**Comment by maxcountryman** [comment]
How do you want to order merging this with #59?

**Comment by weiznich** [comment]
> How do you want to order merging this with https://github.com/maxcountryman/tower-sessions/pull/59?

Merge this and #60 first, and I will rebase #59 afterwards.

**Comment by maxcountryman** [comment]
This is merged now.

**Pull request closed without merging**